src/Ledger/Ledger/Properties.agda

```diff
@@ -128,145 +133,136 @@ isGADeposit dp = isGADepositᵇ dp ≡ true
 govDepsMatch : LState → Set
 govDepsMatch ⟦ utxoSt , govSt , _ ⟧ˡ =
   filterˢ isGADeposit (dom (utxoSt .deposits ))
-  ≡ᵉ fromList (map (λ (id , _) → GovActionDeposit id) govSt)
+  ≡ᵉ fromList (map f govSt)
   where open UTxOState using (deposits)
 
-V⊎P∖V→DepPurposes : {txid : TxId} → List (GovVote ⊎ GovProposal) → List DepositPurpose
-V⊎P∖V→DepPurposes [] = []
-V⊎P∖V→DepPurposes {txid} (inj₁ v ∷ vps) = V⊎P∖V→DepPurposes{txid} vps -- omit votes
-V⊎P∖V→DepPurposes {txid} (inj₂ p ∷ vps) = GovActionDeposit (txid , length vps) ∷ (V⊎P∖V→DepPurposes{txid} vps)
-
 instance _ = +-0-monoid
 
+
 module _
+
+  {gΓ : GovEnv}
+
   -- ASSUMPTIONS (TODO: eliminate these) --
-  {filterGA : ∀ {txid} {n} → filterˢ isGADeposit ❴ GovActionDeposit (txid , n) ❵
-                              ≡ᵉ ❴ GovActionDeposit (txid , n) ❵ }
-  {filterCD : ∀ {c} {pp} → filterˢ isGADeposit (dom ((certDeposit c {pp})ˢ)) ≡ᵉ ∅}
-  {filterCR : (c : DCert) {deps : DepositPurpose ⇀ Coin}
-              → filterˢ isGADeposit (dom ( deps ∣ certRefund c ᶜ ˢ ))
-                ≡ᵉ filterˢ isGADeposit (dom (deps ˢ))}
-  {dom-single : {d : DepositPurpose} {c : Coin} → ❴ d ❵ˢ ≡ dom ❴ d , c ❵ˢ }
-  {ΔgovSt≡ : {txid : TxId} {govSt govSt' : GovState} {txb : TxBody}
-             → fromList (map (λ (id , _) → GovActionDeposit id) govSt')
-               ≡ᵉ fromList (map (λ (id , _) → GovActionDeposit id) govSt)
-                  ∪ fromList (V⊎P∖V→DepPurposes{txid} (txgov txb))}
+  {- 1 -} {dom-single : {d : DepositPurpose} {c : Coin} → ❴ d ❵ˢ ≡ dom ❴ d , c ❵ˢ }
+  {- 2 -} {filterCD   : ∀ {c} {pp} → filterˢ isGADeposit (dom ((certDeposit c {pp})ˢ)) ≡ᵉ ∅}
+  {- 3 -} {filterGA   : ∀ {txid} {n} → filterˢ isGADeposit ❴ GovActionDeposit (txid , n) ❵ ≡ᵉ ❴ GovActionDeposit (txid , n) ❵ }
+  {- 4 -} {filterCR   : (c : DCert) {deps : DepositPurpose ⇀ Coin}
+                        → filterˢ isGADeposit (dom ( deps ∣ certRefund c ᶜ ˢ )) ≡ᵉ filterˢ isGADeposit (dom (deps ˢ))}
   where
 
+  open SetoidReasoning (≡ᵉ-Setoid{DepositPurpose})
+  module ≡ᵉ = IsEquivalence (≡ᵉ-isEquivalence {DepositPurpose})
+
+  -- TODO: replace with std lib functions (using, e.g., using isInj₂, from-just, etc)
+  is-inj₂ : Pred (GovVote ⊎ GovProposal) _
+  is-inj₂ = λ {(inj₁ _) → ⊥; (inj₂ _) → ⊤}
+
+  dec-is-inj₂ : ∀ x → Dec (is-inj₂ x)
+  dec-is-inj₂ = λ {(inj₁ _) → no λ (); (inj₂ _) → yes tt}
+
   LEDGER-govDepsMatch : LedgerInvariant _⊢_⇀⦇_,LEDGER⦈_ govDepsMatch
 
   LEDGER-govDepsMatch
-    (LEDGER-I⋯ tx-not-valid (UTXOW-inductive⋯ _ _ _ _ _ (UTXO-inductive⋯ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ ⊢utxo)))
-    aprioriMatch = case ⊢utxo of λ where
+    (LEDGER-I⋯ tx-not-valid (UTXOW-inductive⋯ _ _ _ _ _ (UTXO-inductive⋯ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ UTXOS-sts)))
```

Comment:
what's the intention here of doing a pattern match and then a case?

Reply:
just to have less duplication of code (less repetition of the patterns)... but much of this will likely change quite a bit when I address Andre's comments re. not needing anything from the expanded UTXOW and UTXO patterns.